build.gradle

```diff
@@ -105,6 +105,7 @@ repositories {
 dependencies {
     implementation "org.opensearch:opensearch:${opensearch_version}"
     implementation 'org.junit.jupiter:junit-jupiter:5.10.0'
+    implementation 'org.projectlombok:lombok:1.18.22'
```

Comment:
We should have a separate discussion/issue/RFC on whether we want to use a code generation tool like Lombok, and figure out what modifications we need to make to our CI to support it.

Reply:
removed lombok dependency for now. It's not needed in this pr. I will open a discuss and figure out the XContent question before adding this dependency